tests/rptest/infinite_retention/checks.py

```diff
@@ -0,0 +1,156 @@
+# Copyright 2023 Redpanda Data, Inc.
+#
+# Use of this software is governed by the Business Source License
+# included in the file licenses/BSL.md
+#
+# As of the Change Date specified in that file, in accordance with
+# the Business Source License, use of this software will be governed
+# by the Apache License, Version 2.0
+
+# noqa: E501
+
+from ast import NodeTransformer, parse, unparse, Constant
+from copy import deepcopy
+
+
+# Credit to blhsing:
+# https://stackoverflow.com/questions/74528551/how-to-substitute-value-of-variables-in-python-expression-but-not-evaluate-the
+class NamesToConstants(NodeTransformer):
+    _args = {}
+
+    def __init__(self, args):
+        self._args = args
+
+    def visit_Name(self, node):
+        # using own dict of vars instead of globals()
+        if node.id in self._args:
+            value = self._args[node.id]
+            # convert value to integer if viable
+            try:
+                value = int(value)
+            except Exception:
+                pass
+            return Constant(value=value)
+        return node
+
+
+class RedpandaChecksBase(object):
```

Comment:
A short comment on the class to say it's purpose would be helpful

Reply:
This is a helper class for transforming precompiled expression to contain constants instead of variable names.
```
"var + anothervar" becomes "23 + 5"
```
The end goal is to understand which values were used in variables when assertion check for something was not successful.

Reply:
See another comment for example.

Reply:
I get it, but I think it's useful to put this in a comment for the benefit of future readers.